### ISSUE TITLE
VS 2015 - Having two adjecent foreach statements with the same "iterator" variable name results in the second one being null.

### ISSUE THREAD

**Issue by cdomino** [dotnet/roslyn#8148]
See the attached file (for references to the numbers used here). Basically, I have two adjacent foreach loops (i.e. not nested, although the second one is protected by a try…catch) (#’s 1 and 2) with the same variable name ("pm" in this case) for the “iterators.” It appears as though there is scoping issue where the first loop is “disposing” the variable by name; it is null in the second loop (#4), although the first item in the collection has a value (#3).  I originally wrote the code using VS 2013, which is why I think it’s something to do with the IDE, not .NET (4.5.1) in both cases. Using a different name for the variable in the second loop works.

![issue](https://cloud.githubusercontent.com/assets/9473687/12540389/da7b96be-c2cd-11e5-92b5-86d95374cf9c.png)

Thanks!

**Comment by tmat** [comment]
Have you made any changes to the code while debugging?

**Comment by cdomino** [comment]
Hey Tomas –

 

Thanks for the quick reply!

 

I first noticed the issue after coming back from a hibernation (I’m on Windows 10, by the way) and I don’t recall if I had explicitly changed the code during that first debugging session. But once I noticed the issue, it remained after cycling VS and even rebooting. That said, this is a large code base, and I make heavy usage of that same pattern (using the same variable name for my iterators) and haven’t noticed anything else. However, I *did* encounter this in two difference places in that same class/file.

 

Let me know if this helps or what else I can tell you.

 

Thanks again!
